scipy/interpolate/_polyint.py

```diff
@@ -210,14 +210,34 @@ def derivative(self, x, der=1):
 
         Notes
         -----
-        This is computed by evaluating all derivatives up to the desired
+        This may be computed by evaluating all derivatives up to the desired
         one (using self.derivatives()) and then discarding the rest.
 
         """
         x, x_shape = self._prepare_x(x)
         y = self._evaluate_derivatives(x, der+1)
         return self._finish_y(y[der], x_shape)
 
+    def _evaluate_derivatives(self, x, der=None):
+        """
+        Actually evaluate the derivatives.
+
+        Parameters
+        ----------
+        x : array_like
+            1D array of points at which to evaluate the derivatives
+        der: integer, optional
+            The number of derivatives to evaluate, from 'order 0' (der=1) to order der-1.
+            If omitted, return all possibly-non-zero derivatives, ie 0 to order n-1.
+
+        Returns
+        -------
+        d : ndarray
+            Array of shape ``(der, x.size, self.yi.shape[1])`` containing
+            the derivatives from 0 to der-1
+        """
+        raise NotImplementedError()
```

Comment:
We should add a message and also test this.

Reply:
What should the test/message be?  Currently it matches the abstract method for `_evaluate()` in `_Interpolator1D` here:
```
def _evaluate(self, x):
        """
        Actually evaluate the value of the interpolator.
        """
        raise NotImplementedError()
```
but I can add a message to/test for both if that's something that needs testing.

Reply:
It looks like the same situation elsewhere in scipy is handled by making the abstract class a subclass of `ABC` and using an `@abstractmethod` decorator, together with:
```
raise NotImplementedError("This method is only implemented with an "
                          "associated child of the base class.")
```
which could be a reasonable message to re-use here.

Reply:
This class is not part of the public API so I think we are fine to leave it as is